File: leet-code/createTargetArray.go
package main

func createTargetArray(nums []int, index []int) []int {
	result := make([]int, len(nums))
	for i, val := range index {
		copy(result[val+1:], result[val:])
		result[val] = nums[i]
	}
	return result
}

File: leet-code/createTargetArray.js
/**
 * @param {number[]} nums
 * @param {number[]} index
 * @return {number[]}
 */
var createTargetArray = function (nums, index) {
  var result = []
  for (let i = 0; i < index.length; i++) {
    for (let j = result.length - 1; j >= index[i]; j--) {
      result[j + 1] = result[j]
    }
    result[index[i]] = nums[i]
  }
  return result
}

console.log(createTargetArray([0, 1, 2, 3, 4], [0, 1, 2, 2, 1]))
